src/python/pants/backend/python/rules/inject_init.py

```diff
@@ -0,0 +1,42 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from pants.backend.python.subsystems.pex_build_util import identify_missing_init_files
+from pants.engine.fs import EMPTY_DIRECTORY_DIGEST, Digest, Snapshot
+from pants.engine.isolated_process import ExecuteProcessRequest, ExecuteProcessResult
+from pants.engine.rules import rule
+from pants.engine.selectors import Get
+from pants.util.objects import datatype
+
+
+class InitInjectedDigest(datatype([('directory_digest', Digest)])): pass
```

Comment:
A very small nit (I feel like I post a lot of those, people should stop listening to them at some point.
This represents the digest of an init file that has been injected, so I think `InjectedInitDigest` is a slightly better name.

Feel free to ignore.

Also, a comment at the top would be nice:
```suggestion
# This represents the digest of an init file that has been injected into a module.
class InitInjectedDigest(datatype([('directory_digest', Digest)])): pass
```

Reply:
Great suggestion! Once we close #7710 and can return the merged original digest + injected digest, we should rename this back to `InitInjectedDigest`. For now, `InjectedInitDigest` is a much better name.

> I feel like I post a lot of those, people should stop listening to them at some point.

Please don't! They are always really helpful changes. The details, like what to name things, matter with programming.